Persisted Queries: Avoid adding to non time related screens

psealock opened the pull request:
Fixes https://github.com/woocommerce/woocommerce-admin/issues/2214
Fixes https://github.com/woocommerce/woocommerce-admin/issues/2176

Some screens don't have a datepicker. Navigating to those pages now don't persist time-related url params.

This PR also introduces Node at LTS for our build process. This is what [Gutenberg](https://github.com/WordPress/gutenberg/blob/master/.nvmrc) uses.

### Detailed test instructions:

1. Go to Dashboard.
2. Choose a date range.
3. Inspect wc-admin urls. With the exception of devdocs, stock, settings, and customers pages, all urls should have a time related query, eg `?period=last_week&compare=previous_year`.
4. Inspect other urls to make sure no regressions occurred.

psealock commented on client/layout/controller.js:
```diff
@@ -124,6 +127,12 @@ class Controller extends Component {
+		'stock',
+		'settings',
+		'customers',
+	] );
```
@rrennick, you'll want to add `scheduled-actions` to this array in your plugin so that time-related queries don't get added to the url for the `Scheduled Actions` link in the sidebar. 

For example, scheduled actions sidebar link may look like this:

```
http://tangaroa.test/wp-admin/admin.php?page=wc-admin#/analytics/scheduled-actions?period=last_week&compare=previous_year
```

jeffstieler replied:
Woah, @rrennick do you have a Scheduled Actions report built on this codebase?

jeffstieler replied:
✨ https://github.com/Prospress/action-scheduler-admin ✨

rrennick replied:
I also need to update it for the new page registering code.

psealock commented:
Thanks @timmyc 

TIL regex negative lookbehinds are not supported by all browsers. I simplified the regex and removed the lookbehind. Its ready for another review.

Aljullu commented on client/layout/controller.js:
```diff
@@ -121,28 +124,55 @@ class Controller extends Component {
+ */
+export function updateLinkHref( item, nextQuery, excludedScreens ) {
+	/**
+	 * Regular expression with a lookbehind for "page=wc-admin", matching any WooCommerce Admin screen.
```
@psealock should this comment be updated now that you removed the lookbehind?

psealock replied:
Fixed in https://github.com/woocommerce/woocommerce-admin/pull/2225/commits/a25746bb59480104abdd3360e0f5ef359820fd1d

psealock commented:
>But I found a behavior that I'm not sure if it's correct

The behaviour you're seeing is expected for the scope of this PR, but brings up an interesting point.

#2250 Is to set a default range, which IMO means that when no parameters are in the query, a default is chosen. For now, the default is hard coded to "Month to date" and #2250 calls for allowing the use to choose instead.

What you're proposing here is to persist the time related queries (whether default or otherwise) to not just the next screen but any subsequent screens until changed.

@LevinMedia, what do you think about this^^^. My opinion is to follow through with #2250 and see if we get any feedback, but we can easily store the query and retrieve it too.
